samza-core/src/test/java/org/apache/samza/runtime/TestApplicationRunnerMain.java

```diff
@@ -79,6 +82,28 @@ public void TestStatusOperation() throws Exception {
     assertEquals(1, TestApplicationRunnerInvocationCounts.statusCount);
   }
 
+  @Test
+  public void TestLoadConfig() {
+    ApplicationRunnerMain.ApplicationRunnerCommandLine cmdLine = new ApplicationRunnerMain.ApplicationRunnerCommandLine();
+    OptionSet options = cmdLine.parser().parse(
+        "--config-loader-factory",
+        "org.apache.samza.config.loaders.PropertiesConfigLoaderFactory",
+        "--config-loader-properties",
+        "path=" + getClass().getResource("/test.properties").getPath(),
+        "-config", String.format("%s=%s", ApplicationConfig.APP_CLASS, MockStreamApplication.class.getName()),
+        "-config", String.format("app.runner.class=%s", TestApplicationRunnerInvocationCounts.class.getName()));
+
+    Config actual = cmdLine.loadConfig(options);
+
+    assertEquals(4, actual.size());
+    assertEquals("org.apache.samza.config.loaders.PropertiesConfigLoaderFactory", actual.get(JobConfig.CONFIG_LOADER_FACTORY));
+    assertEquals(
+        getClass().getResource("/test.properties").getPath(),
+        actual.get(ConfigLoaderFactory.CONFIG_LOADER_PROPERTIES_PREFIX + "path"));
+    assertEquals(MockStreamApplication.class.getName(), actual.get(ApplicationConfig.APP_CLASS));
+    assertEquals(TestApplicationRunnerInvocationCounts.class.getName(), actual.get("app.runner.class"));
```

Comment:
Minor suggestion: Instead of needing to check the size and each individual entry, could you build the expected config map and then do an equals check?

Reply:
Good point, updated.